bin/packer.py

```diff
@@ -64,33 +65,37 @@ def contains(x, ys):
                 return False
         return True
 
-    with open(aws_config) as fh:
-        cred = json.load(fh)
-        session = Session(aws_access_key_id = cred["aws_access_key"],
-                          aws_secret_access_key = cred["aws_secret_key"],
-                          region_name = 'us-east-1')
-
-        client = session.client('ec2')
-        response = client.describe_images(Filters=[
-                        {"Name": "owner-id", "Values": ["099720109477"]},
-                        {"Name": "virtualization-type", "Values": ["hvm"]},
-                        {"Name": "root-device-type", "Values": ["ebs"]},
-                        {"Name": "architecture", "Values": ["x86_64"]},
-                        #{"Name": "platform", "Values": ["Ubuntu"]},
-                        #{"Name": "name", "Values": ["hvm-ssd"]},
-                        #{"Name": "name", "Values": ["14.04"]},
-                   ])
-
-        images = response['Images']
-        images = [i for i in images if contains(i['Name'], ('hvm-ssd', '14.04', 'server'))]
-        images.sort(key=lambda x: x["CreationDate"], reverse=True)
-
-        if len(images) == 0:
-            print("Error: could not locate base AMI, exiting ....")
-            sys.exit(1)
+    if aws_config == None:
+        session = aws.use_iam_role()
+
+    else:
+        with open(aws_config) as fh:
+            cred = json.load(fh)
+            session = Session(aws_access_key_id = cred["aws_access_key"],
+                            aws_secret_access_key = cred["aws_secret_key"],
+                            region_name = 'us-east-1')
+
+    client = session.client('ec2')
+    response = client.describe_images(Filters=[
+                    {"Name": "owner-id", "Values": ["099720109477"]},
```

Comment:
Not your code, but where does this owner id come from?

Reply:
Hmmm I'm not really sure...

Reply:
It is the official Ubuntu owner id, so that we get official Ubuntu images (and not from someone else)